docs/sources/migration-guide/migrating-from-thanos-or-prometheus.md

```diff
@@ -78,15 +78,254 @@ If you need to have external labels in your query results, this is currently not
+   level=info ts=2022-10-10T13:04:59.649337602Z caller=compactor.go:42 msg="processed 30.00% of 701243 series"
+   level=info ts=2022-10-10T13:05:02.986219042Z caller=compactor.go:42 msg="processed 40.00% of 701243 series"
+   level=info ts=2022-10-10T13:05:05.990498497Z caller=compactor.go:42 msg="processed 50.00% of 701243 series"
+   level=info ts=2022-10-10T13:05:09.349918024Z caller=compactor.go:42 msg="processed 60.00% of 701243 series"
+   level=info ts=2022-10-10T13:05:12.040895624Z caller=compactor.go:42 msg="processed 70.00% of 701243 series"
+   level=info ts=2022-10-10T13:05:15.253899238Z caller=compactor.go:42 msg="processed 80.00% of 701243 series"
+   level=info ts=2022-10-10T13:05:18.471471014Z caller=compactor.go:42 msg="processed 90.00% of 701243 series"
+   level=info ts=2022-10-10T13:05:21.536267363Z caller=compactor.go:42 msg="processed 100.00% of 701243 series"
+   level=info ts=2022-10-10T13:05:21.536466158Z caller=tools_bucket.go:1222 msg="wrote new block after modifications; flushing" source=01GEGWPME2187SVFH63G8DH7KH new=01GF0ZWPWGEPHG5NV79NH9KMPV
+   level=info ts=2022-10-10T13:05:28.675240198Z caller=tools_bucket.go:1231 msg="uploading new block" source=01GEGWPME2187SVFH63G8DH7KH new=01GF0ZWPWGEPHG5NV79NH9KMPV
+   level=info ts=2022-10-10T13:05:38.922348564Z caller=tools_bucket.go:1241 msg=uploaded source=01GEGWPME2187SVFH63G8DH7KH new=01GF0ZWPWGEPHG5NV79NH9KMPV
+   level=info ts=2022-10-10T13:05:38.979696873Z caller=block.go:203 msg="block has been marked for deletion" block=01GEGWPME2187SVFH63G8DH7KH
+   level=info ts=2022-10-10T13:05:38.979832767Z caller=tools_bucket.go:1249 msg="rewrite done" IDs=01GEGWPME2187SVFH63G8DH7KH
+   level=info ts=2022-10-10T13:05:38.980197796Z caller=main.go:161 msg=exiting
+   ```
+   
+   
+   Cleanup the original blocks which are marked for deletion.
+  
+   ```bash
+   thanos tools bucket cleanup \
+       --objstore.config-file bucket.yaml \
+       --delete-delay=0
+   ```
+  
+   > **Note**: If there are multiple prometheus clusters, then relabelling each of them in parallel may speed up the entire process.
+   > Get the list of blocks that for each cluster, and process it separately.
+   > ```bash
+   > thanos tools bucket inspect \
+   >     --objstore.config-file bucket.yaml \
+   >     --output=tsv \
+   >     | grep <PROMETHEUS-CLUSTER-NAME> \
+   >     | awk '{print $1}' > prod-blocks.tsv
+   >```
+  
+   >```bash
+   > for ID in `cat prod-blocks.tsv`
+   > do
+   >     thanos tools bucket rewrite  \
+   >        --objstore.config-file bucket.yaml \
+   >        --rewrite.to-relabel-config-file relabel-config.yaml \
+   >        --delete-blocks \
+   >        --no-dry-run \
+   >        --id $ID         
+   > done
+   >```
+ 
+5) Remove external labels from meta.json
+   
+   Mimir compactor will not be able to compact the blocks with external labels in their meta.json. Therefore these external labels have to be  removed before copying them to the Mimir bucket.
```

Comment:
This isn't quite correct. Mimir compactor can compact blocks with external labels just fine, but only with other blocks using the same external labels. Since Mimir's own blocks don't have any such labels, we remove labels from these imported blocks too.